webpack.config.js

```diff
@@ -110,6 +111,42 @@ var wpconfig = {
                 use: 'babel-loader'
             },
             {
+                test: /\.(js|jsx)$/,
+                include: [
+                    /studio-frontend/,
+                    /paragon/
+                ],
+                use: 'babel-loader'
+            },
```

Comment:
Oooh, actually, I think you could do this in an even simpler way. If both `/studio-frontend/` and `/paragon/` are nested within `node_modules`, you could change the `exclude` on line 108 to something like `/node_modules\/(?!(paragon|studio-frontend)\/).*/` -- then you wouldn't need the includes. (I'm not 100% sure that'll work, but it worked on my old paragon integration PR (https://github.com/edx/edx-platform/pull/15282/files#diff-11e9f7f953edc64ba14b0cc350ae7b9dR114)

Reply:
Yeah, this is the one that I had to wrestle with quite a bit :/

I tried placing the exclude and include in the same loader, that caused issues. Love the regex idea though, I'll try that out.

Reply:
So I've learned quite a bit about webpack loaders from all this. Placing include and exclude in the same loader does not work because exclude [takes preferrence over test and include](https://webpack.js.org/configuration/). The regex approach works, but only when you've installed studio-frontend via npm. When installing locally in devstack, we run into the issue webpack warns about here: https://webpack.js.org/configuration/module/#rule-conditions

```
Be careful! The resource is the resolved path of the file, which means symlinked resources are the real path not the symlink location. This is good to remember when using tools that symlink packages (like npm link), common conditions like /node_modules/ may inadvertently miss symlinked files.
```

What ends up happening is that the *top level* modules (`edx-platform/node_modules/@edx/studio-frontend` and `edx-platform/node_modules/@edx/paragon`) are included, but the locally-installed version of studio-frontend contains a *nested* `node_modules/@edx/paragon`, whose full expanded path ends up being `/edx/src/sfe_parent/studio-frontend/~/@edx/paragon`. We *must* add an explicit rule to grab this file, as the "exclude" rule is only looking for files under `/edx/app/exapp/edx-platform`.

I was able to get a solution working with the "exclude" regex and a near-duplicate "issuer" rule, but I've elected to go back to this iteration (no regex, explicit `include` rule) for simplicity's sake.

The only downside I see is that we're now webpacking a bit more than we need to (see below) in dev mode, but I don't think that will cause any issues. 

`[../../../src/sfe_parent/studio-frontend/node_modules/react-dom/index.js] /edx/src/sfe_parent/studio-frontend/~/react-dom/index.js 58 bytes {0} [built]`

Reply:
This is representative of the errors I was seeing with just the regexed "exclude" rule:

```
ERROR in /edx/src/sfe_parent/studio-frontend/~/@edx/paragon/src/Button/index.jsx
Module parse failed: /edx/src/sfe_parent/studio-frontend/node_modules/@edx/paragon/src/Button/index.jsx Unexpected token (17:4)
You may need an appropriate loader to handle this file type.
|     onKeyDown,
|     type,
|     ...other
|   } = props;
| 
 @ /edx/src/sfe_parent/studio-frontend/src/BackendStatusBanner/index.jsx 23:14-48
 @ /edx/src/sfe_parent/studio-frontend/src/index.jsx
```

Reply:
Thanks for the context! This is all super good to know -- sorry for sending you down this rabbit hole. The explicit `include` seems a lot easier to understand.